src/main/webapp/2Dscript.js

```diff
@@ -414,40 +476,67 @@ function placeRectangleEnd() {
   if (infoWindow !== undefined) {
     infoWindow.setMap(null);
   }
-  // Clear other paths.
-  // line
-  // minLatPoint
-  // lat , lng
-  // minLngPoint
-  // lat, lng
-  const subSectionData = computeSubSection(currentRun.data,
-      currentLat, priorLat, currentLng, priorLng);
-  // Choose a subSectionNumber, implement differently in future pr.
-  const subSectionNumber = 1;
-  // Clear prior paths, only display newly selected ones.
-  if (markerBottom !== undefined) {
-    markerBottom.setMap(null);
-  }
-  if (markerTop !== undefined) {
-    markerTop.setMap(null);
+  clearSelectedPaths(['subSection', 'markerBottom', 'markerTop']);
+  // Clear other paths. line minLatPoint lat , lng minLngPoint lat, lng Iterate
+  // over all data values here, need to find a way to check the checkbox value,
+  // should likely store this in an object somewhere to prevent excess DOM
+  // queries.
+  const mapRuns = Object.entries(runs);
+  for ([id, currentRun] of mapRuns) {
+    if (currentRun.checkBox.checked) {
+      const subSectionData = computeSubSection(currentRun.data,
+          currentLat, priorLat, currentLng, priorLng);
+      currentRun.subLine = subSectionData.subLine;
+      // Clear prior paths, only display newly selected ones.
+      currentRun.subSection = getPolyLine(subLine, 'blue', 1.0, 2);
+      // Setup event listener to show option for 3D window when polyline is
+      // clicked.
+      google.maps.event.addListener(subPath, 'click', linkTo3D);
+      subPath.setMap(map);
+      currentRun.markerBottom = genMarker(subSectionData.minLatPoint.lat,
+          subSectionData.minLatPoint.lng);
+      currentRun.markerTop = genMarker(subSectionData.maxLngPoint.lat,
+          subSectionData.maxLngPoint.lng);
+      showAll([currentRun.subSection, currentRun.markerBottom,
+        currentRun.markerTop]);
+      isMouseDown = false;
```

Comment:
Can you move this line out into the outer event handler? The causality is more clear there. Same for `isMouseDown = true` in `placeRectangleStart`.